src/wallet/wallet.cpp

```diff
@@ -1336,13 +1336,29 @@ void CWallet::MarkConflicted(const uint256& hashBlock, int conflicting_height, c
     if (conflictconfirms >= 0)
         return;
 
-    // Do not flush the wallet here for performance reasons
+    auto get_updated_state = [&](CWalletTx& wtx) {
+        LOCK(cs_wallet);
+
+        if (conflictconfirms < GetTxDepthInMainChain(wtx)) {
+            wtx.m_state = TxStateConflicted{hashBlock, conflicting_height};
+            return true;
+        }
+        return false;
+    };
+
+    RecursiveUpdateTxState(hashTx, get_updated_state);
+
+}
+
+void CWallet::RecursiveUpdateTxState(const uint256& tx_hash, const GetUpdatedStateFn& get_updated_state) {
```

Comment:
In c3fe92175b6ac709e4fdfd4766253c75d7ffd184 "wallet: introduce generic recursive tx state updating function"

This could also be used by `AbandonTransaction`.